src/lib/sidenav/sidenav.ts

```diff
@@ -175,53 +154,28 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
   }
 
 
-  /** Open this sidenav, and return a Promise that will resolve when it's fully opened (or get
-   * rejected if it didn't). */
-  open(): Promise<MdSidenavToggleResult> {
+  /**  Open the sidenav. */
+  open(): void {
     return this.toggle(true);
   }
 
-  /**
-   * Close this sidenav, and return a Promise that will resolve when it's fully closed (or get
-   * rejected if it didn't).
-   */
-  close(): Promise<MdSidenavToggleResult> {
+  /** Close the sidenav. */
+  close(): void {
     return this.toggle(false);
   }
 
   /**
-   * Toggle this sidenav. This is equivalent to calling open() when it's already opened, or
-   * close() when it's closed.
+   * Toggle this sidenav.
    * @param isOpen Whether the sidenav should be open.
-   * @returns Resolves with the result of whether the sidenav was opened or closed.
    */
-  toggle(isOpen: boolean = !this.opened): Promise<MdSidenavToggleResult> {
-    // Shortcut it if we're already opened.
-    if (isOpen === this.opened) {
-      return this._toggleAnimationPromise ||
-          Promise.resolve(new MdSidenavToggleResult(isOpen ? 'open' : 'close', true));
-    }
+  toggle(isOpen: boolean = !this.opened): void {
```

Comment:
I'm just asking since I don't know much about the animations API, how do I as a user who is currently doing this:

```ts
snav.toggle().then(() => doStuff())
```
change it to work after this PR? I would guess something like this?

```html
<md-sidenav (@transform.start)="doStuff()"></md-sidenav>
```